File: app/src/main/java/com/smartassistant/overlay/NotificationTipManager.java
package com.smartassistant.overlay;

import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.util.Log;

import androidx.core.app.NotificationCompat;

/**
 * Manages gaming tip notifications as an alternative to overlay bubbles
 */
public class NotificationTipManager {
    private static final String TAG = "NotificationTipManager";
    
    private static final String CHANNEL_ID = "GAMING_TIPS_CHANNEL";
    private static final String CHANNEL_NAME = "نصائح الألعاب";
    private static final String CHANNEL_DESC = "نصائح وإرشادات ذكية أثناء اللعب";
    
    private static final int NOTIFICATION_ID_BASE = 2000;
    
    private Context context;
    private NotificationManager notificationManager;
    private int notificationCounter = 0;
    
    public NotificationTipManager(Context context) {
        this.context = context;
        this.notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
        createNotificationChannel();
    }
    
    private void createNotificationChannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            NotificationChannel channel = new NotificationChannel(
                CHANNEL_ID,
                CHANNEL_NAME,
                NotificationManager.IMPORTANCE_HIGH // High importance for gaming tips
            );
            
            channel.setDescription(CHANNEL_DESC);
            channel.enableLights(true);
            channel.enableVibration(false); // Don't vibrate during gaming
            channel.setShowBadge(true);
            channel.setLockscreenVisibility(Notification.VISIBILITY_PUBLIC);
            
            notificationManager.createNotificationChannel(channel);
            Log.d(TAG, "Notification channel created");
        }
    }
    
    public void showGameTip(String tipText, String tipType) {
        if (tipText == null || tipText.isEmpty()) return;
        
        // Create dismiss intent
        Intent dismissIntent = new Intent(context, NotificationDismissReceiver.class);
        PendingIntent dismissPendingIntent = PendingIntent.getBroadcast(
            context, 
            notificationCounter, 
            dismissIntent, 
            PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE
        );
        
        // Build notification
        NotificationCompat.Builder builder = new NotificationCompat.Builder(context, CHANNEL_ID)
            .setSmallIcon(getTipIcon(tipType))
            .setContentTitle(getTipTitle(tipType))
            .setContentText(tipText)
            .setStyle(new NotificationCompat.BigTextStyle().bigText(tipText))
            .setPriority(NotificationCompat.PRIORITY_HIGH)
            .setCategory(NotificationCompat.CATEGORY_RECOMMENDATION)
            .setAutoCancel(true)
            .setTimeoutAfter(8000) // Auto-dismiss after 8 seconds
            .setDeleteIntent(dismissPendingIntent)
            .setOngoing(false)
            .setShowWhen(false);
        
        // Add color based on tip type
        builder.setColor(getTipColor(tipType));
        
        // Create unique notification ID
        int notificationId = NOTIFICATION_ID_BASE + (notificationCounter % 10);
        notificationCounter++;
        
        // Show notification
        try {
            notificationManager.notify(notificationId, builder.build());
            Log.d(TAG, "Game tip notification shown: " + tipText);
        } catch (Exception e) {
            Log.e(TAG, "Error showing tip notification", e);
        }
    }
    
    private int getTipIcon(String tipType) {
        // Use system icons based on tip type
        switch (tipType != null ? tipType : BubbleOverlayService.TIP_TYPE_GENERAL) {
            case BubbleOverlayService.TIP_TYPE_ATTACK:
                return android.R.drawable.ic_media_play;
            case BubbleOverlayService.TIP_TYPE_DEFENSE:
                return android.R.drawable.ic_menu_close_clear_cancel;
            case BubbleOverlayService.TIP_TYPE_ELIXIR:
                return android.R.drawable.ic_dialog_alert;
            case BubbleOverlayService.TIP_TYPE_TIME:
                return android.R.drawable.ic_lock_idle_alarm;
            default:
                return android.R.drawable.ic_dialog_info;
        }
    }
    
    private String getTipTitle(String tipType) {
        switch (tipType != null ? tipType : BubbleOverlayService.TIP_TYPE_GENERAL) {
            case BubbleOverlayService.TIP_TYPE_ATTACK:
                return "🔥 نصيحة هجومية";
            case BubbleOverlayService.TIP_TYPE_DEFENSE:
                return "🛡️ نصيحة دفاعية";
            case BubbleOverlayService.TIP_TYPE_ELIXIR:
                return "⚡ حالة الإكسير";
            case BubbleOverlayService.TIP_TYPE_TIME:
                return "⏰ تنبيه وقت";
            default:
                return "🎮 نصيحة ذكية";
        }
    }
    
    private int getTipColor(String tipType) {
        switch (tipType != null ? tipType : BubbleOverlayService.TIP_TYPE_GENERAL) {
            case BubbleOverlayService.TIP_TYPE_ATTACK:
                return 0xFFFF4444; // Red for attack
            case BubbleOverlayService.TIP_TYPE_DEFENSE:
                return 0xFF4444FF; // Blue for defense
            case BubbleOverlayService.TIP_TYPE_ELIXIR:
                return 0xFFFF8800; // Orange for elixir
            case BubbleOverlayService.TIP_TYPE_TIME:
                return 0xFFFFAA00; // Yellow for time
            default:
                return 0xFF44AA44; // Green for general
        }
    }
    
    public void clearAllTips() {
        try {
            // Cancel all tip notifications
            for (int i = 0; i < 10; i++) {
                notificationManager.cancel(NOTIFICATION_ID_BASE + i);
            }
            Log.d(TAG, "All tip notifications cleared");
        } catch (Exception e) {
            Log.e(TAG, "Error clearing tip notifications", e);
        }
    }
    
    /**
     * Receiver for handling notification dismissal
     */
    public static class NotificationDismissReceiver extends android.content.BroadcastReceiver {
        @Override
        public void onReceive(Context context, Intent intent) {
            Log.d(TAG, "Tip notification dismissed by user");
        }
    }
}

File: app/src/main/kotlin/com/smartassistant/overlay/NotificationTipManager.kt
package com.smartassistant.overlay

import android.app.Notification
import android.app.NotificationChannel
import android.app.NotificationManager
import android.app.PendingIntent
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import android.util.Log
import androidx.core.app.NotificationCompat

/**
 * Manages gaming tip notifications as an alternative to overlay bubbles
 */
class NotificationTipManager(private val context: Context) {
    
    companion object {
        private const val TAG = "NotificationTipManager"
        private const val CHANNEL_ID = "GAMING_TIPS_CHANNEL"
        private const val CHANNEL_NAME = "نصائح الألعاب"
        private const val CHANNEL_DESC = "نصائح وإرشادات ذكية أثناء اللعب"
        private const val NOTIFICATION_ID_BASE = 2000
    }
    
    private val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
    private var notificationCounter = 0
    
    init {
        createNotificationChannel()
    }
    
    private fun createNotificationChannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val channel = NotificationChannel(
                CHANNEL_ID,
                CHANNEL_NAME,
                NotificationManager.IMPORTANCE_HIGH // High importance for gaming tips
            ).apply {
                description = CHANNEL_DESC
                enableLights(true)
                enableVibration(false) // Don't vibrate during gaming
                setShowBadge(true)
                lockscreenVisibility = Notification.VISIBILITY_PUBLIC
            }
            
            notificationManager.createNotificationChannel(channel)
            Log.d(TAG, "Notification channel created")
        }
    }
    
    fun showGameTip(tipText: String?, tipType: String?) {
        if (tipText.isNullOrEmpty()) return
        
        // Create dismiss intent
        val dismissIntent = Intent(context, NotificationDismissReceiver::class.java)
        val dismissPendingIntent = PendingIntent.getBroadcast(
            context,
            notificationCounter,
            dismissIntent,
            PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
        )
        
        // Build notification
        val builder = NotificationCompat.Builder(context, CHANNEL_ID)
            .setSmallIcon(getTipIcon(tipType))
            .setContentTitle(getTipTitle(tipType))
            .setContentText(tipText)
            .setStyle(NotificationCompat.BigTextStyle().bigText(tipText))
            .setPriority(NotificationCompat.PRIORITY_HIGH)
            .setCategory(NotificationCompat.CATEGORY_RECOMMENDATION)
            .setAutoCancel(true)
            .setTimeoutAfter(8000) // Auto-dismiss after 8 seconds
            .setDeleteIntent(dismissPendingIntent)
            .setOngoing(false)
            .setShowWhen(false)
            .setColor(getTipColor(tipType))
        
        // Create unique notification ID
        val notificationId = NOTIFICATION_ID_BASE + (notificationCounter % 10)
        notificationCounter++
        
        // Show notification
        try {
            notificationManager.notify(notificationId, builder.build())
            Log.d(TAG, "Game tip notification shown: $tipText")
        } catch (e: Exception) {
            Log.e(TAG, "Error showing tip notification", e)
        }
    }
    
    private fun getTipIcon(tipType: String?): Int {
        // Use system icons based on tip type
        return when (tipType ?: BubbleOverlayService.TIP_TYPE_GENERAL) {
            BubbleOverlayService.TIP_TYPE_ATTACK -> android.R.drawable.ic_media_play
            BubbleOverlayService.TIP_TYPE_DEFENSE -> android.R.drawable.ic_menu_close_clear_cancel
            BubbleOverlayService.TIP_TYPE_ELIXIR -> android.R.drawable.ic_dialog_alert
            BubbleOverlayService.TIP_TYPE_TIME -> android.R.drawable.ic_lock_idle_alarm
            else -> android.R.drawable.ic_dialog_info
        }
    }
    
    private fun getTipTitle(tipType: String?): String {
        return when (tipType ?: BubbleOverlayService.TIP_TYPE_GENERAL) {
            BubbleOverlayService.TIP_TYPE_ATTACK -> "🔥 نصيحة هجومية"
            BubbleOverlayService.TIP_TYPE_DEFENSE -> "🛡️ نصيحة دفاعية"
            BubbleOverlayService.TIP_TYPE_ELIXIR -> "⚡ حالة الإكسير"
            BubbleOverlayService.TIP_TYPE_TIME -> "⏰ تنبيه وقت"
            else -> "🎮 نصيحة ذكية"
        }
    }
    
    private fun getTipColor(tipType: String?): Int {
        return when (tipType ?: BubbleOverlayService.TIP_TYPE_GENERAL) {
            BubbleOverlayService.TIP_TYPE_ATTACK -> 0xFFFF4444.toInt() // Red for attack
            BubbleOverlayService.TIP_TYPE_DEFENSE -> 0xFF4444FF.toInt() // Blue for defense
            BubbleOverlayService.TIP_TYPE_ELIXIR -> 0xFFFF8800.toInt() // Orange for elixir
            BubbleOverlayService.TIP_TYPE_TIME -> 0xFFFFAA00.toInt() // Yellow for time
            else -> 0xFF44AA44.toInt() // Green for general
        }
    }
    
    fun clearAllTips() {
        try {
            // Cancel all tip notifications
            repeat(10) { i ->
                notificationManager.cancel(NOTIFICATION_ID_BASE + i)
            }
            Log.d(TAG, "All tip notifications cleared")
        } catch (e: Exception) {
            Log.e(TAG, "Error clearing tip notifications", e)
        }
    }
    
    /**
     * Receiver for handling notification dismissal
     */
    class NotificationDismissReceiver : BroadcastReceiver() {
        override fun onReceive(context: Context, intent: Intent) {
            Log.d(TAG, "Tip notification dismissed by user")
        }
    }
}